packages/terra-date-input/tests/wdio/date-input-spec.js

```diff
@@ -405,7 +441,7 @@ Terra.describeViewports('Date Input', ['medium'], () => {
         Terra.hideInputCaret('input[name="terra-date-year-date-input"]');
 
         browser.click('input[name="terra-date-year-date-input"]');
-        browser.keys(['1', 'ArrowDown']);
+        browser.keys(['7654']);
```

Comment:
This visuals here shows 765, should an warning message show or if the first number is larger than the first digit in the min/max year should it allow you to enter it at all?

Reply:
cc @benbcai , your thoughts?

Reply:
I think it would be difficult to handle "if the first number is larger than the first digit in the min/max year should it allow you to enter it at all?". The reason being the first digit entered may not always represent the first digit in the intended. For example, the user may enter `8` first and use the left arrow key to then enter `201` with the intention to enter the year 2018. It would seem annoying to prevent entering 8 as the first digit in this case.

This test case resembles when the user is pasting `7654` in the input. I think it is more ideal and less confusing to prevent the entire paste event instead of allowing it but just paste the first 3 digits `765` and cutting off the last digit because it would exceed the max year. Pasting may not be a very common use case here let alone pasting an invalid value. So I don't have a strong opinion here.